api/v1beta2/ibmpowervscluster_types.go

```diff
@@ -81,6 +119,27 @@ type IBMPowerVSClusterList struct {
 	Items           []IBMPowerVSCluster `json:"items"`
 }
 
+// VPCResource holds the VPC information.
+type VPCResource struct {
+	// Name of VPC
+	// +optional
+	Name string `json:"name"`
+
+	// IBM Cloud VPC zone where the cluster's loadbalancers will be created
+	Zone string `json:"vpcZone"`
+
+	// Subnet contains the details about subnet
+	Subnet Subnet `json:"subnet"`
+
+	// TransitGateway holds the resource details of TransitGateway
+	TransitGateway TransitGateway `json:"transitGateway"`
```

Comment:
I believe TransitGateway itself is a different component and works like connector between the vpc and classic n/w, so wondering if we need to define outside the vpc data structures